cmds/fixmynetboot/main.go

```diff
@@ -1,22 +1,105 @@
 
-func main() {
+var dhcpErrorBanner = `
+**************************************************************************
+** DHCPv6 failed, see the output above to debug the issue.               *
+** Entering the emergency shell, where you can run "fixmynetboot" again, *
+** or any other LinuxBoot command.                                       *
+**************************************************************************
+`
+
+var (
+	doEmergencyShell = flag.Bool("shell", false, "Run emergency shell if checks fail")
+)
+
+func checkInterface(ifname string) error {
 	checklist := []checker.Check{
-		checker.Check{"wlp2s0 exists", checker.InterfaceExists("wlp2s0"), nil, false},
-		checker.Check{"eth0 exists", checker.InterfaceExists("eth0"), checker.InterfaceRemediate("eth0"), false},
-		checker.Check{"eth0 link speed", checker.LinkSpeed("eth0", 100), nil, false},
-		checker.Check{"eth0 link autoneg", checker.LinkAutoneg("eth0", true), nil, false},
-		checker.Check{"eth0 has link-local", checker.InterfaceHasLinkLocalAddress("wlp2s0"), nil, false},
-		checker.Check{"eth0 has global addresses", checker.InterfaceHasGlobalAddresses("wlp2s0"), nil, false},
+		checker.Check{
+			Name:        fmt.Sprintf("%s exists", ifname),
+			Run:         checker.InterfaceExists(ifname),
+			Remediate:   checker.InterfaceRemediate(ifname),
+			StopOnError: true,
+		},
+		checker.Check{
+			Name:        fmt.Sprintf("%s link speed", ifname),
+			Run:         checker.LinkSpeed(ifname, 100),
+			Remediate:   nil,
+			StopOnError: false},
+		checker.Check{
+			Name:        fmt.Sprintf("%s link autoneg", ifname),
+			Run:         checker.LinkAutoneg(ifname, true),
+			Remediate:   nil,
+			StopOnError: false,
+		},
+		checker.Check{
+			Name:        fmt.Sprintf("%s has link-local", ifname),
+			Run:         checker.InterfaceHasLinkLocalAddress(ifname),
+			Remediate:   nil,
+			StopOnError: true,
+		},
+		checker.Check{
+			Name:        fmt.Sprintf("%s has global addresses", ifname),
+			Run:         checker.InterfaceHasGlobalAddresses("eth0"),
+			Remediate:   nil,
+			StopOnError: true,
+		},
+	}
+
+	return checker.Run(checklist)
+}
+
+func getNonLoopbackInterfaces() ([]string, error) {
+	var interfaces []string
+	allInterfaces, err := net.Interfaces()
+	if err != nil {
+		return nil, err
 	}
+	for _, iface := range allInterfaces {
+		if iface.Flags&net.FlagLoopback == 0 {
+			interfaces = append(interfaces, iface.Name)
+		}
+	}
+	return interfaces, nil
+}
 
-	checker.Run(checklist)
+func main() {
+	flag.Parse()
+	var (
+		interfaces []string
+		err        error
+	)
+	ifname := flag.Arg(0)
+	if ifname == "" {
+		interfaces, err = getNonLoopbackInterfaces()
+		if err != nil {
+			log.Fatal(err)
+		}
+	} else {
+		interfaces = []string{ifname}
+	}
 
+	for _, ifname := range interfaces {
+		if err := checkInterface(ifname); err != nil {
+			if !*doEmergencyShell {
+				log.Fatal(err)
+			}
+			if err := checker.EmergencyShell(dhcpErrorBanner)(); err != nil {
```

Comment:
Is this dhcp banner still relevant?

Reply:
Not anymore, I'll leave it out

Reply:
actually I'll just change the message to be generic rather than about DHCP

Reply:
Should the `dhcp` name also be dropped?

Reply:
done